more fleet framework migrations

daemitus opened the pull request:
- add a fleen gen transform to de-pointer `*[]sometype` and `*map[string]sometype` fields
- migrate resource fleet_agent_policy
- migrate resource fleet_integration
- migrate data_source fleet_integration
- migrate resource fleet_server_host

tobio commented on internal/fleet/agent_policy/resource_test.go:
Can we add an acceptance test here (and for the other resources) verifying state compatibility between the SDK and FW resources ([example](https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/elasticsearch/index/index/acc_test.go#L80-L175)).

tobio commented on internal/fleet/integration/update.go:
```diff
@@ -0,0 +1,39 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func (r *integrationResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```
Can we share the `Create` implementation here instead?

tobio commented on internal/fleet/agent_policy/resource_test.go:
```diff
@@ -57,7 +57,7 @@ func TestAccResourceAgentPolicy(t *testing.T) {
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"skip_destroy"},
+				ImportStateVerifyIgnore: []string{"monitor_metrics", "monitor_logs", "skip_destroy"},
```
What's happening here? This PR populates these fields on `Read`, why are we not setting them to computed?

daemitus commented:
I was having problems earlier testing those, I think it has to do with on
read, not knowing if they should be false or null. I'll try that bit again
with a computerd/default false, and on read set false if not set.

Regarding the x-pointers flag, I can revert that. Will just need to change
up the helpers to do pointers instead. As you said, it doesn't change much.

As you've brought up manipulation of the published spec, whats your take
then on all of the nested anonymous structs? Fleet isn't too bad, but
Kibana's is a disaster for generation at the moment.

On Thu, Sep 26, 2024, 11:26 PM Toby Brain ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In internal/fleet/agent_policy/resource_test.go
> <https://github.com/elastic/terraform-provider-elasticstack/pull/785#discussion_r1777962451>
> :
>
> > @@ -57,7 +57,7 @@ func TestAccResourceAgentPolicy(t *testing.T) {
>  				ResourceName:            "elasticstack_fleet_agent_policy.test_policy",
>  				ImportState:             true,
>  				ImportStateVerify:       true,
> -				ImportStateVerifyIgnore: []string{"skip_destroy"},
> +				ImportStateVerifyIgnore: []string{"monitor_metrics", "monitor_logs", "skip_destroy"},
>
> What's happening here? This PR populates these fields on Read, why are we
> not setting them to computed?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/terraform-provider-elasticstack/pull/785#pullrequestreview-2332679631>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFMTBCAIQCRDCIZZRAEOD3ZYTGABAVCNFSM6AAAAABOR4KYFKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGMZSGY3TSNRTGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> .com>
>

daemitus commented:
ok. 
- x-go-pointer flag is dropped.
- fixed agent policy monitoring test ignores finally
- shared integration.create with update
- added FromSDK tests to all 3 migrated resources

tobio commented:
> whats your take then on all of the nested anonymous structs?

Yea, these are a total pain. Ultimately we need to fix these in the source though so anyone using the specs can benefit.

tobio reviewed (approved):
❤️ thanks for doing this work. It simplifies ongoing development loads!

daemitus commented:
And here I couldn't sleep this morning. Looks like I'll be rebasing the
other pr now.

On Sun, Sep 29, 2024, 6:36 AM Toby Brain ***@***.***> wrote:

> Merged #785
> <https://github.com/elastic/terraform-provider-elasticstack/pull/785>
> into main.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/terraform-provider-elasticstack/pull/785#event-14444932539>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFMTBBUUAZ3CAOFV6TBUXDZY7J2VAVCNFSM6AAAAABOR4KYFKVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJUGQ2DIOJTGI2TGOI>
> .
> You are receiving this because you authored the thread.Message ID:
> <elastic/terraform-provider-elasticstack/pull/785/issue_event/14444932539@
> github.com>
>
